node/pkg/libp2p/libp2p.go

```diff
@@ -239,41 +281,54 @@ func initDHT(ctx context.Context, h host.Host, bootstrap string) *dht.IpfsDHT {
 	}
 	wg.Wait()
 
-	return kademliaDHT
+	return kademliaDHT, nil
+}
+
+func connectToPeers(ctx context.Context, h host.Host, routingDiscovery *drouting.RoutingDiscovery, topicName string) (connected bool, err error) {
+	peerChan, err := routingDiscovery.FindPeers(ctx, topicName)
+	if err != nil {
+		return false, err
+	}
+	var wg sync.WaitGroup
+	for p := range peerChan {
```

Comment:
In this loop, we receive multiple (1+) peer IDs, and if we can connect to at least one of them, `connected` variable becomes `true` which is then returned after going through all observed peers. Is this expected behavior? Shouldn't we log potential connection errors?

Reply:
intention was to immediately return true if peer with same topic string is found. I'll make an update

Reply:
So it means we are okay some of them fail to connect?

Reply:
yes